File: tests/test_parse/data/sample.pas
program UserManager;

type
  TUser = record
    Name: string;
    Age: Integer;
  end;

function IsAdult(User: TUser): Boolean;
begin
  Result := User.Age >= 18;
end;

procedure ProcessUser(User: TUser);
begin
  if IsAdult(User) then
    WriteLn('Processing adult user: ', User.Name)
  else
    WriteLn('Processing minor user: ', User.Name);
end;

begin
  // Main program
end. 

File: tests/test_parse/data/sample.f90
module user_manager
  implicit none
  
  type :: user
    character(len=20) :: name
    integer :: age
  end type user
  
contains
  logical function is_adult(u)
    type(user), intent(in) :: u
    is_adult = u%age >= 18
  end function is_adult
end module user_manager 